lib/scanner/walk.go

```diff
@@ -412,11 +413,12 @@ func (w *walker) walkDir(ctx context.Context, relPath string, info fs.FileInfo,
 		if cf.IsEquivalent(f, w.IgnorePerms, true) {
 			return nil
 		}
-		if cf.Invalid {
-			// We do not want to override the global version with the file we
-			// currently have. Keeping only our local counter makes sure we are in
-			// conflict with any other existing versions, which will be resolved by
-			// the normal pulling mechanisms.
+		if cf.IsInvalid() {
+			// The old file was invalid for whatever reason and probably not
+			// up to date with what was out there in the cluster. Drop all
+			// others from the version vector to indicate that we haven't
+			// taken their version into account, and possibly cause a
+			// conflict.
```

Comment:
Shouldn't MustRescan be excempt from this? If I understand correctly if a file is changed on disk and a request happens before the scan, the MustRescan bit gets set. In that case, we don't want to generate a conflict, we just want to force rescan the file and update the version as normal.  
I should shorten my review intervals...

Reply:
I made a note about that somewhere, but it may not have survived. My initial thinking was exactly this, as well. But - we only set the MustRescan bit when we are 1) about to pull/commit a file while pulling and 2) it's changed on disk compared to the index. This is per definition a conflict already so handling it the same way here shouldn't change the behavior.

Reply:
1. I can't find that, the only usage `SetMustRescan` I see is when requesting: https://github.com/syncthing/syncthing/pull/4952/files#diff-1df9805b2381b6288b220226f82d5dc9R1397

2. Why is it a conflict when it changes on disk compared to what's in the index, i.e. I don't see the difference to a normal change of file on disk by user interaction.

Reply:
You're right as usual. Please consider the updated version!